### ISSUE TITLE
Better accuracy with smaller block in S3DIS

### ISSUE THREAD

**Issue by josegemez** [aRI0U/RandLA-Net-pytorch#42]
Hi, I have implemented this repository with custom S3DIS data (xyz and normal only) I had tried with entire rooms, and blocks (5x5, 2x2, and 1x1)... The best accuracy I have got is with the smallest block... Is it possible that local encoder not work properly? It can related with the square distances? Thanks in advance. 
![2022_09_28@8-53](https://user-images.githubusercontent.com/67518595/192716084-7809ebb7-b30b-4ee5-a51c-6cfd74fd9643.PNG)

**Comment by josegemez** [comment]
I had got some improvements... I had mix some code from:

[https://github.com/isl-org/Open3D-ML/tree/master/ml3d/torch/models](https://github.com/isl-org/Open3D-ML/tree/master/ml3d/torch/models
)
![image](https://user-images.githubusercontent.com/67518595/195531207-272929d0-54a7-4b68-ad6b-dff758c3cd6f.png)

**Comment by josegemez** [comment]
More improvements... I have tweaked dropout in the last mlp and the batch normalization parameters...

![image](https://user-images.githubusercontent.com/67518595/196005684-bb35d7f3-4ca1-46d6-a6a4-6fc0d1cb5916.png)

**Comment by Dlemand** [comment]
Thank you very much. i also find that the net cannot learn the features.....May be I should learn something from OPEN3D.

**Comment by ashishrana160796** [comment]
Hi @josegemez, I tried similar improvements with Open3D's code, even though the baseline results were better, but still I am only getting good scores for classes with larger point clouds only while inputting whole room as input. Except for few classes, for most of the other classes the model is learning nothing.

Could you please provide more specific details on the exact input samples during the training and your code improvements from the Open3D-ML's implementation. It would be really helpful, thanks in advance!

**Comment by KiKuBoy** [comment]
@josegemez Is it convenient to open source your code? I used the code of openml to merge and there is still no good result, thank you

**Comment by jscksy** [comment]
@xiaoxina12 同学你好，你能跑通这个代码吗，我这边一直遇到各种bug

**Comment by KiKuBoy** [comment]
![2023-01-09 17-19-28屏幕截图](https://user-images.githubusercontent.com/74182024/211275724-8b5dc17e-2f5a-47c6-94e6-cef1314b6ece.png)
可以跑通，但是根本学不到特征，我又对照其他各个版本改了下，还是很难收敛。

**Comment by jscksy** [comment]
同学能开源或者发到我邮箱405357455@qq.com吗，非常感谢？我刚入门，一路debug还是没跑通
![image](https://user-images.githubusercontent.com/31052472/211277390-2bc6fff8-c940-455d-a721-5acd04da6ec3.png)

**Comment by KiKuBoy** [comment]
@jscksy 可以的，等晚上吧，我的代码也没有收敛，期待你可以把它收敛。

**Comment by jscksy** [comment]
> 

感谢！我就是个愁毕设的小菜鸟q_q

**Comment by SC-shendazt** [comment]
> @jscksy 可以的，等晚上吧，我的代码也没有收敛，期待你可以把它收敛。

能加个好友吗，我改了一份代码是哪个自动驾驶版本的，也是遇到了问题，想发一份给你一起交流我的微信shendaln

**Comment by jscksy** [comment]
可以啊，我后面跑出来的也挺奇怪的，我的邮箱 ***@***.***




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "aRI0U/RandLA-Net-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月8日(星期三) 晚上9:29
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [aRI0U/RandLA-Net-pytorch] Better accuracy with smaller block in S3DIS (Issue #42)





  
@jscksy 可以的，等晚上吧，我的代码也没有收敛，期待你可以把它收敛。
  
能加个好友吗，我改了一份代码是哪个自动驾驶版本的，也是遇到了问题，想发一份给你一起交流

**Comment by mostafa501** [comment]
@josegemez 
Dear josegemez, may you help me provide your modified code, as this one learns no features, and has a very bad results.
could you help me, this is my email: mostafahanafy311@gmail.com.
thanks in advance

**Comment by Xuelangshuo** [comment]
> 



> @jscksy 可以的，等晚上吧，我的代码也没有收敛，期待你可以把它收敛。

同学你好，请问可以发我一份代码吗，我最近也在复现这个，遇到很多bug，我的邮箱是674001266@qq.com，万分感谢

**Comment by wangzihao77** [comment]
> 更多改进...我调整了最后一个mlp中的退出和批量标准化参数......
> 
> ![图像](https://user-images.githubusercontent.com/67518595/196005684-bb35d7f3-4ca1-46d6-a6a4-6fc0d1cb5916.png)

**Comment by wangzihao77** [comment]
> ![2023-01-09 17-19-28屏幕截图](https://user-images.githubusercontent.com/74182024/211275724-8b5dc17e-2f5a-47c6-94e6-cef1314b6ece.png)
> 可以跑通，但是根本学不到特征，我又对照其他各个版本改了下，还是很难收敛。

好兄弟，你的代码能分享给我一份吗，我的结果还没你的高，我的邮箱是：2762311960@qq.com

**Comment by wangzihao77** [comment]
@xiaoxina12

**Comment by wangzihao77** [comment]
@josegemez 好兄弟，可以分享一份你改进的代码吗，我复现出的结果也是挺糟糕的，我的邮箱是：[2762311960@qq.com](mailto:2762311960@qq.com)

**Comment by wangzihao77** [comment]
> ![2023-01-09 17-19-28屏幕截图](https://user-images.githubusercontent.com/74182024/211275724-8b5dc17e-2f5a-47c6-94e6-cef1314b6ece.png)
> 可以跑通，但是根本学不到特征，我又对照其他各个版本改了下，还是很难收敛。

你好，可以分享下你的代码吗？

**Comment by didiwei60** [comment]
@josegemez 
Dear josegemez, 
may you help me provide your modified code. 
Very bad results were gotten when using this code.
Could you please help me,  this is my email: 971144518@qq.com
And thank you very much

**Comment by 667825a** [comment]
> ![2023-01-09 17-19-28屏幕截图](https://user-images.githubusercontent.com/74182024/211275724-8b5dc17e-2f5a-47c6-94e6-cef1314b6ece.png)
> 可以跑通，但是根本学不到特征，我又对照其他各个版本改了下，还是很难收敛。

可以发我一份代码吗，我的邮箱778913421@qq.com

**Comment by 667825a** [comment]
> 同学能开源或者发到我邮箱[405357455@qq.com](mailto:405357455@qq.com)吗，非常感谢？我刚入门，一路debug还是没跑通
> ![image](https://user-images.githubusercontent.com/31052472/211277390-2bc6fff8-c940-455d-a721-5acd04da6ec3.png)

我也出现这个问题了，最后怎么解决的呢
